app/services/twilio_api_service.rb

```diff
@@ -23,6 +23,16 @@ class TwilioApiService
   delegate :logger, to: Rails
 
   class Error < StandardError
+    # https://www.twilio.com/docs/api/errors
+    ERROR_CODE_REASONS = {21211 => :invalid_phone_number,
+                          21614 => :invalid_phone_number}
+
```

Comment:
This handles individual twilio errors now instead of relying on the http response code.